components/script/dom/request.rs

```diff
@@ -419,46 +419,43 @@ impl Request {
         r.request.borrow_mut().headers = r.Headers(can_gc).get_headers_list();
 
         // Step 34. Let inputBody be input’s request’s body if input is a Request object; otherwise null.
-        let mut input_body = if let RequestInfo::Request(ref mut input_request) = input {
+        let input_body = if let RequestInfo::Request(ref mut input_request) = input {
             let mut input_request_request = input_request.request.borrow_mut();
+            r.body_stream.set(input_request.body().as_deref());
```

Comment:
This was the missing line. After I had updated the whole constructor to be aligned with the spec, the test still wasn't working. Then I discovered that for `init` we were correctly setting the stream, but for `input` we weren't.